src/services/images/getMapillaryImage.ts

```diff
@@ -58,17 +58,20 @@ const getMapillaryImageRaw = async (poiCoords: Position): Promise<Image> => {
     return { ...item, angleFromPhotoToPoi, deviationFromStraightSight };
   });
 
-  const sorted = photos.sort(
+  const panos = photos.filter((pic) => pic.is_pano);
+  const sorted = (panos.length ? panos : photos).sort(
     (a, b) => a.deviationFromStraightSight - b.deviationFromStraightSight,
   );
 
   debugOutput(sorted);
 
+  const imageToUse = sorted[0]; // it is save to assume that it has at least one element
   return {
     source: 'Mapillary',
     link: `https://www.mapillary.com/app/?focus=photo&pKey=${sorted[0].id}`,
-    thumb: sorted[0].thumb_1024_url,
-    timestamp: new Date(sorted[0].captured_at).toLocaleString(),
+    thumb: imageToUse.thumb_1024_url,
```

Comment:
btw, when i click the fullscreen button on the viewer, the image is pretty pixelated. Is there perhaps any option to use some HD version for panos?

But this one is not a blocker to merge this PR. Feel free to leave it 🙂

Reply:
Is it ok the way I just did it?

Panorama images get the full original image. Normal images still only the 1024p image

Reply:
Yes, thank you very much. I really love this feature :) merging it right away 👍